examples/hooks-react-ts/src/models/dolphins.ts

```diff
@@ -5,7 +5,7 @@ import { RootModel } from '.';
 export const dolphins = createModel<RootModel>()({
 	state: 0,
 	reducers: {
-		increment: (state) => state + 1,
+		increment: (state, payload: number = 1) => state + payload,
```

Comment:
Could we leave the `: number` annotation?

Reply:
without `:number` annotation, the type of `payload` would be `any`. Because:

![image](https://user-images.githubusercontent.com/1198651/100084976-6d595b00-2e86-11eb-9030-f125f18ee8a3.png)

Reply:
Yes exactly,  I also have some doubts here: TS should know the payload type by assigning a numeric value to it but it couldn't... 

Do you have any ideas bro?

Reply:
I am not sure if it was an [issue](https://github.com/microsoft/TypeScript/issues/41668) with TS. 

And I also created a [playground](https://www.typescriptlang.org/play?ts=4.2.0-dev.20201124#code/KYDwDg9gTgLgBAE2AYwDYEMrDjAnmbAWQiVQCVgEBXZYKAZwB4AVAZRnRmwF450A7XAD44vAN4BYAFABIANoBrYLgBccejCgBLfgHMAumorVaUFu07Ah0gL7TpoSLEQoMWHPmzGadcxy6ifIIivAAU0jIalmps-sAANBFg6LioEOgIAPxqArjSAJSiIrGW9lLIEPwacMhYlsSkaqGF3CKMEWRwoFz8CPRwDcDklD4MjKxCibKs0kKhALYQapKy-OjzwNnqmjq6EVFcajOyWCZ09Ftkti0iKzJrG1sa2nr7cUcRp6P0RraBzUU4AsIPkcoJASs4HAsDAqFB+HAxAA6FGLOxSdHSCpVeC1YD1EhDADqWhgAAsIFQYN5TOxtMgYFpKj8gTc4O1ZGQpjJjnNFssIg9Nmpnrs3tE4McZF9TBdfhi2XchU8dq9ZAdgB8TiNZfKbP82cDQUFcBDpFCYXCEcjURB0ZipAB6R1wZKpdIIV2Ydb9HRwAAG9GAMFYEA25N2-pqAjg-Ag8AARtgdAAzOiUPj9fhUeZJqBY5nwFMQCCBPEE0jNUKQ7YSgCMUwtOvOAqkUKhQZDYeDZN2oQ1amzubo8S97oygTrhRr7ehwatta4AG5ze30VD7fkys6x2kJ8koD64H7A8HQ+He3oo8gY0nj-w06dM7Gc3mCzi4AnMGW6lxBqgSXJSlqWbKA6S0BkmSqKsawHOAG1XGUW0RVcOzPbsIz0ft3hfYcoFHN0909XgpxQttZznWF4UXYAV3IjdV03bcXUIj0vUPeZfQRU8uwvSNowRO9U3TT10CzV86DgAB3UkyS6cBUAg0kPAIII438KD32qL8AC8f3xP9CVQGDVzghDyKQhhW1nTtzx7Ps4KHPMCJSIjBwkqBJ2nVD20taiNTo2d1zgJipCAA) which you can have a look if you are interested.

Reply:
There isn't support for partial type inference yet. There are some workarounds like currying https://github.com/microsoft/TypeScript/pull/26349, so probably this should stay like this until Typescript fixes this :)

Reply:
Yes, my issue marked as `design limitation`. I will merge this